res/scripts/celmi/timetables/timetable_helper.lua

```diff
@@ -1,558 +1,568 @@
-        local res = {}
-        for k, v in pairs(lineObject.stops) do
-            res[k] = v.stationGroup
-        end
-        return res
-    else
-        return {}
-    end
-end
-
----@param station number | string
--- returns stationName : String
-function timetableHelper.getStationName(station)
-    if type(station) == "string" then station = tonumber(station) end
-    if not(type(station) == "number") then return "ERROR" end
-
-    local err, res = pcall(function()
-        return api.engine.getComponent(station, api.type.ComponentType.NAME)
-    end)
-    if err and res then return res.name else return "ERROR" end
-end
-
-
----@param line number | string
----@param stationNumber number
--- returns stationID : Number and -1 in Error Case
-function timetableHelper.getStationID(line, stationNumber)
-    if type(line) == "string" then line = tonumber(line) end
-    if not(type(line) == "number") then return -1 end
-
-    local lineObject = api.engine.getComponent(line, api.type.ComponentType.LINE)
-    if lineObject and lineObject.stops and lineObject.stops[stationNumber] then
-        return lineObject.stops[stationNumber].stationGroup
-    else
-        return -1
-    end
-end
-
--------------------------------------------------------------
----------------------- Array Functions ----------------------
--------------------------------------------------------------
-
----@param arr table
--- returns [Number], an Array where the index it the source element and the number is the target position
-function timetableHelper.getOrderOfArray(arr)
-    local toSort = {}
-    for k,v in pairs(arr) do
-        toSort[k] = {key =  k, value = v}
-    end
-    table.sort(toSort, function(a,b)
-        return string.lower(a.value) < string.lower(b.value)
-    end)
-    local res = {}
-    for k,v in pairs(toSort) do
-        res[k-1] = v.key-1
-    end
-    return res
-end
-
----@param a table
----@param b table
--- returns Array, the merged arrays a,b
-function timetableHelper.mergeArray(a,b)
-    if a == nil then return b end
-    if b == nil then return a end
-    local ab = {}
-    for _, v in pairs(a) do
-        table.insert(ab, v)
-    end
-    for _, v in pairs(b) do
-        table.insert(ab, v)
-    end
-    return ab
-end
-
-
--- returns [{vehicleID: lineID}]
-function timetableHelper.getAllVehiclesAtTerminal()
-    return api.engine.system.transportVehicleSystem.getVehiclesWithState(api.type.enum.TransportVehicleState.AT_TERMINAL)
-    --[[local res = {}
-    local vehicleMap = api.engine.system.transportVehicleSystem.getLine2VehicleMap()
-    for k,v in pairs(vehicleMap) do
-        if (hasTimetable(k)) then
-            for _,v2 in pairs(v) do
-                res[v2] = k
-            end
-        end
-    end
-    return res]]--
-end
-
-
--- returns Number, current GameTime in seconds
-function timetableHelper.getTime()
-    local time = api.engine.getComponent(api.engine.util.getWorld(), api.type.ComponentType.GAME_TIME).gameTime
```

Comment:
From reading the transportfever.net forum I now understand that calls to the API should be done on separate lines. Does this also extend to the `api.type.ComponentType.[type_name]`?

Also I am not sure if the `.gameTime` should be split to a separate line as well. See https://github.com/IncredibleHannes/TPF2-Timetables/issues/50
```
local timeComponent = api.engine.getComponent(apiWorld, apiGameTime)
local time = timeComponent.gameTime
```

Reply:
I would not change it, the ways its now. This is only a call of this object not call up of the api itself.